tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -1071,17 +1067,19 @@ class MklConvOp : public OpKernel {
       // TODO(mdfaijul): Extend the basic parameters for data types and fusions
       this->ExtendConvFwdParams(context, convFwdDims);
 
-      conv_fwd = MklConvFwdPrimitiveFactory<float, Tinput, Tfilter, Tbias,
-                                            Ttemp_output>::Get(convFwdDims,
-                                                               do_not_cache);
+      conv_fwd =
+          MklConvFwdPrimitiveFactory<Tinput, Tfilter, Tbias, Ttemp_output>::Get(
+              convFwdDims, do_not_cache);
 
       // Allocate output tensors `output_tensor` and `filter_out_tensor`
       std::shared_ptr<ConvFwdPd> conv_fwd_pd = conv_fwd->GetPrimitiveDesc();
       AllocateOutputTensor(context, *conv_fwd_pd, dst_dims_mkl_order, tf_fmt,
                            &dst_tensor);
       Tensor* filter_out_tensor = nullptr;
-      if (typeid(Tinput) == typeid(float) && typeid(Tfilter) == typeid(float) &&
-          typeid(Toutput) == typeid(float)) {
+      if (typeid(Tinput) == typeid(Tfilter) &&
+          typeid(Tinput) == typeid(Toutput) &&
+          (typeid(Tinput) == typeid(float) ||
+           typeid(Tinput) == typeid(bfloat16))) {
```

Comment:
This set of conditions appears twice. (The first time in line 975.) How about storing it in a bool?

Reply:
Sure. Addressed.